resources/views/home/_search_result_forum_post.blade.php

```diff
@@ -0,0 +1,66 @@
+{{--
+    Copyright 2015-2018 ppy Pty. Ltd.
+
+    This file is part of osu!web. osu!web is distributed with the hope of
+    attracting more community contributions to the core ecosystem of osu!.
+
+    osu!web is free software: you can redistribute it and/or modify
+    it under the terms of the Affero GNU General Public License version 3
+    as published by the Free Software Foundation.
+
+    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+    See the GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+--}}
+{{-- more code than template in this view :best: --}}
+@php
+    $users = $search->users()->select('user_id', 'username', 'user_avatar')->get();
+@endphp
+
+@foreach ($search->data() as $entry)
+    @php
+        // $entry should be of type App\Libraries\Elasticsearch\Hit
+        $innerHits = $entry->innerHits('posts');
+        $firstPost = $entry->innerHits('first_post'); // instance of App\Libraries\Elasticsearch\SearchResponse
+        $firstPostUrl = route('forum.topics.show', $entry->source('topic_id'));
+        $excerpt = implode('', array_map(function ($post) {
+            return html_excerpt($post->source('search_content'));
+        }, iterator_to_array($firstPost)));
+
+        $user = $users->where('user_id', $entry->source('poster_id'))->first() ?? new App\Models\DeletedUser();
```

Comment:
thankfully there shouldn't be too many users otherwise looping the whole thing repeatedly would be a bit slow.

Reply:
When this was initially written, I was hoping the implementation would be smart enough since it didn't do any extra queries. Then, I actually looked at the implementation 🥉 

The issue will be less the number of users and more that it's running `array_filter`on every iteration